lib/plugins/after/metrics/middlewareStyle.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var assert = require('assert-plus');
+
+var getAuditPayload = require('./../getAuditPayload');
+
+var UrlStore = require('./UrlStore');
+var TimerStore = require('./TimerStore');
+var StatusCodeStore = require('./StatusCodeStore');
+
+var DEFAULT_MAX_REQUESTS = 1000;
+
+///--- API
+
+/**
+ * The HttpMetrics should be added as an 'after' event on the server.
+ * After each event the metrics will store that event and kick out
+ * any old requests.  Metrics will be delivered on an interval of
+ * options.rate (ms, defaults to 1000).
+ *
+ * @public
+ * @function httpMetrics
+ * @param {Object} options -
+ * @param {Object} options.server -
+ * @param {Array.<Store>} [options.stores] -
+ * @param {Number} [options.limit=1000] -
+ * @param {Number} [options.rate=1000] - How often the server should emit a
+ * httpMetrics event.  Value is in milliseconds
+ * @param {Number} [options.port=8080] - Used in the output server for
+ * what port to listen on.
+ * @param {Boolean} [options.createOutputServer=true] - If an output/debug
+ * server should be created.
+ * @returns {HttpMetrics} -
+ */
+var HttpMetrics = function HttpMetrics(options) {
+    assert.object(options, 'options');
+    assert.object(options.server, 'options.server');
+    assert.optionalNumber(options.limit, 'options.limit');
+    assert.optionalNumber(options.rate, 'options.rate');
+    assert.optionalArray(options.stores, 'options.stores');
+
+    // We do not concern ourselves about 0 (falsy) since that is
+    // truly an silly amount of stored requests.
+    if (!options.limit) {
+        options.limit = DEFAULT_MAX_REQUESTS;
+    }
+
+    // The stores for the metrics.
+    var stores = options.stores || [
+        new UrlStore(),
+        new TimerStore(),
+        new StatusCodeStore()
+    ];
+
+    var storedIdx = -1;
+    var limit = options.limit;
+    var storedPayloads = [];
+    var server = this.server = options.server;
+    var rate = options.rate || 1000;
+
+    function report() {
+        return stores
+            .map(function (store) {
+                return store.report();
+            })
+            .reduce(function (acc, data) {
+                acc[data.name] = data;
+                return acc;
+            }, {});
+    }
+
+    var id = setInterval(function () {
```

Comment:
I think this can be a good approach as then the implementor can distribute the metrics how they choose and it keeps this plugin like the rest.